### PR TITLE
Refactor libafl_qemu creating the Emulator struct and post syscall hooks

### PR DESCRIPTION
**Pull request by andreafioraldi**

### CODE REVIEW

File: libafl_qemu/src/lib.rs [hunk 1]
```diff
@@ -1,40 +1,23 @@
 use std::env;
 
-#[cfg(feature = "aarch64")]
+#[cfg(cpu_target = "aarch64")]
 pub mod aarch64;
```
Comment (domenukk) [inline libafl_qemu/src/lib.rs:4]: `cpu_target` is wrong here, you need the arch of the guest, not the host

Comment (andreafioraldi) [reply libafl_qemu/src/lib.rs:4]: it is

Comment (domenukk) [reply libafl_qemu/src/lib.rs:4]: How? `cpu_target` will always be that of the host, so if I try to emulate aarch64 on X86, I will have only x86 registers available(?)

Comment (domenukk) [reply libafl_qemu/src/lib.rs:4]: https://github.com/AFLplusplus/LibAFL/blob/d484fd1f6d16176df9709d6899a49463a0bf2c90/libafl_qemu/build.rs#L67

That is the answer?

Sorry for my confusion then, there are similar named cfgs inbuilt

Comment (andreafioraldi) [reply libafl_qemu/src/lib.rs:4]: Nope it's target_arch the rust one, cpu_target comes from qemu

**Pull request merged**